pkg/util/resizefs/resizefs_linux.go

```diff
@@ -84,3 +100,19 @@ func (resizefs *ResizeFs) xfsResize(deviceMountPath string) (bool, error) {
 	resizeError := fmt.Errorf("resize of device %s failed: %v. xfs_growfs output: %s", deviceMountPath, err, string(output))
 	return false, resizeError
 }
+
+// findBlockDeviceRescanPath Find the underlaying disk for a linked path such as /dev/disk/by-path/XXXX or /dev/mapper/XXXX
+// will return /sys/devices/pci0000:00/0000:00:15.0/0000:03:00.0/host0/target0:0:1/0:0:1:0/rescan
+func findBlockDeviceRescanPath(path string) (string, error) {
+	devicePath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return "", err
+	}
+	// if path /dev/hdX split into "", "dev", "hdX" then we will
+	// return just the last part
+	parts := strings.Split(devicePath, "/")
+	if len(parts) == 3 && strings.HasPrefix(parts[1], "dev") {
+		return filepath.EvalSymlinks(filepath.Join("/sys/block", parts[2], "device", "rescan"))
+	}
+	return "", fmt.Errorf("Illegal path for device " + devicePath)
```

Comment:
minor nit, I think beginning error messages with capital letter causes some warnings...

Reply:
I copied this text from the https://github.com/kubernetes/kubernetes/blob/d8a420c0d810dafdae85356db3a905e48e732927/pkg/volume/util/device_util_linux.go#L66 as well